internal/controllers/inventory_controller.go

```diff
@@ -1541,6 +1600,47 @@ func (t *reconcilerTask) deployServer(ctx context.Context, serverName string) (u
 		},
 	}
 
+	if utils.IsOAuthEnabled(t.object) && utils.RequiresInternalProxy(serverName) {
```

Comment:
Same comment about having this in its own function.

Reply:
fixed